src/include/planner/operator/simple/logical_simple.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "planner/operator/logical_operator.h"
+
+namespace kuzu {
+namespace planner {
+
+class LogicalSimple : public LogicalOperator {
+public:
+    LogicalSimple(LogicalOperatorType operatorType,
+        std::shared_ptr<binder::Expression> outputExpression)
+        : LogicalOperator{operatorType}, outputExpression{std::move(outputExpression)} {}
+    LogicalSimple(LogicalOperatorType operatorType,
+        std::vector<std::shared_ptr<LogicalOperator>> plans,
+        std::shared_ptr<binder::Expression> outputExpression)
+        : LogicalOperator{operatorType, std::move(plans)},
+          outputExpression{std::move(outputExpression)} {}
+
+    void computeFactorizedSchema() override {
+        createEmptySchema();
+        schema->createGroup();
+        schema->insertToGroupAndScope(outputExpression, 0);
+    }
+
+    void computeFlatSchema() override {
+        createEmptySchema();
+        auto groupPos = schema->createGroup();
+        schema->insertToGroupAndScope(outputExpression, groupPos);
+        schema->setGroupAsSingleState(groupPos);
+    }
+
+    inline std::shared_ptr<binder::Expression> getOutputExpression() const {
```

Comment:
remove inline across this PR. compiler should inline automatically if you write implementation in header.